src/libraries/Native/Unix/CMakeLists.txt

```diff
@@ -152,15 +156,15 @@ endif(CLR_CMAKE_TARGET_FREEBSD)
 #       ./build-native.sh cmakeargs -DCLR_ADDITIONAL_COMPILER_OPTIONS=<...> cmakeargs -DCLR_ADDITIONAL_LINKER_FLAGS=<...>
 #
 if(CLR_CMAKE_TARGET_UNIX)
-    if (CLR_CMAKE_TARGET_DARWIN)
-        add_definitions(-DTARGET_DARWIN)
-        add_link_options(-Wl,-bind_at_load)
-    elseif (CLR_CMAKE_TARGET_ARCH_WASM)
-        # No object stripping for WASM
-    else (CLR_CMAKE_TARGET_DARWIN)
-        add_compile_options($<$<COMPILE_LANGUAGE:ASM>:-Wa,--noexecstack>)
-        add_link_options(-Wl,--build-id=sha1 -Wl,-z,relro,-z,now)
-    endif(CLR_CMAKE_TARGET_DARWIN)
+    if(NOT CLR_CMAKE_TARGET_ARCH_WASM AND NOT CLR_CMAKE_TARGET_IOS)
+        if(CLR_CMAKE_TARGET_OSX)
+            add_definitions(-DTARGET_OSX)
+            add_link_options(-Wl,-bind_at_load)
```

Comment:
All CI jobs (plus my local ios build for macOS) were green with these options. If we are aware of problems it causes, shall we have some test which fails the CI so it doesn't regress? Otherwise, if it has no effect on functionality, maybe we can avoid branching out.

Reply:
We're still bringing up tests for iOS and it will likely take a few more weeks. I checked in a baseline version of the iOS work that was tested manually so I wanted to minimize the changes to it until we can properly do CI testing.